src/services/IPRService.ts

```diff
@@ -72,57 +72,59 @@ export class IPRService {
-				return true;
-			} else {
-				// Process all events except AUTH_DELETE_ACCOUNT when no record exists.
-				return false;
-			}
-		} catch (e: any) {
-			this.logger.error({ message: "getSessionById - failed executing get from dynamodb:", e });
-			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Error retrieving Session");
-		}
-	}
-
-	async saveEventData(userId: string, updateExpression: string, expressionAttributeValues: any): Promise<string | void> {
-
-		this.logger.info({ message: "Saving event data to dynamodb", tableName: this.tableName });
-		const updateSessionInfoCommand = new UpdateCommand({
-			TableName: this.tableName,
-			Key: {
-				userId,
-			},
-			UpdateExpression: updateExpression,
-			ExpressionAttributeValues: expressionAttributeValues,
-		});
+    try {
+        this.logger.info({ message: "Checking if record is flagged for deletion or already processed", tableName: this.tableName });
+        const getSessionCommand = new GetCommand({
+            TableName: this.tableName,
+            Key: {
+                userId,
+            },
+        });
+
+        const session = await this.dynamo.send(getSessionCommand);
+        const eventAttribute = this.eventAttributeMap.get(eventType);
+
+        if (
+            (eventType === TXMA_EVENT_DETAILS.DELETE_ACCOUNT.Name && (!session.Item || session.Item.accountDeletedOn)) ||
+            (session.Item && (session.Item.accountDeletedOn || session.Item[eventAttribute!]))
+        ) {
+            this.logger.info({
+                message: `Record is flagged for deletion or ${eventAttribute} is already set`,
+            });
+            return true;
+        }
+
+        this.logger.info({ message: "Record is not flagged for deletion or processed" });
+        return false;
+    } catch (error) {
+        this.logger.error({ message: "isFlaggedForDeletionOrEventAlreadyProcessed - Error retrieving session from DynamoDB", error });
+        throw new AppError(HttpCodesEnum.SERVER_ERROR, "Error retrieving Session");
+    }
+}
 
-		this.logger.info("Updating session record" );
 
-		try {
-			await this.dynamo.send(updateSessionInfoCommand);
-		} catch (e: any) {
-			this.logger.error({ message: "Failed to update session record in dynamo", e });
-			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Error updating session record");
-		}
+	async saveEventData(userId: string, UpdateExpression: string, ExpressionAttributeValues: any): Promise<string | void> {
+    try {
+        this.logger.info({ message: "Saving event data to DynamoDB", tableName: this.tableName });
+        const updateSessionInfoCommand = new UpdateCommand({
+            TableName: this.tableName,
+            Key: {
+                userId,
+            },
+            UpdateExpression,
+            ExpressionAttributeValues,
+        });
+
+        this.logger.info("Updating session record");
```

Comment:
We log the `tableName` in the `isFlaggedForDeletionOrEventAlreadyProcessed` function do you think that might be helpful here too?

Reply:
We log it on line 108 within this same func